homeassistant/auth/auth_store.py

```diff
@@ -239,16 +243,29 @@ def async_get_refresh_token_by_token(
 
         return found
 
+    @callback
+    def async_get_refresh_tokens(self) -> list[models.RefreshToken]:
+        """Get all refresh tokens."""
+        return list(
+            itertools.chain.from_iterable(
+                user.refresh_tokens.values() for user in self._users.values()
+            )
+        )
+
     @callback
     def async_log_refresh_token_usage(
         self, refresh_token: models.RefreshToken, remote_ip: str | None = None
     ) -> None:
         """Update refresh token last used information."""
         refresh_token.last_used_at = dt_util.utcnow()
         refresh_token.last_used_ip = remote_ip
+        if refresh_token.expire_at:
+            refresh_token.expire_at = (
+                refresh_token.last_used_at.timestamp() + REFRESH_TOKEN_EXPIRATION
+            )
         self._async_schedule_save()
 
-    async def async_load(self) -> None:
+    async def async_load(self) -> None:  # noqa: C901
```

Comment:
`C901 `async_load` is too complex (26 > 25)`

as alternative, we could remove this one, than we would be back to `25` complexity 😁 
```python
            # Filter out the old keys that don't have jwt_key (pre-0.76)
            if "jwt_key" not in rt_dict:
                continue
```

Reply:
Add the C901. We can clean this up later.  We shouldn't do a cleanup in this PR as they can be a bit riskier and it's always good to have the option to easily revert that type of change